src/GraphQL/Types/ResolveFieldContext.cs

```diff
@@ -79,42 +79,22 @@ public ResolveFieldContext(ResolveFieldContext context)
 
         public TType GetArgument<TType>(string name, TType defaultValue = default)
         {
-            return (TType)GetArgument(typeof(TType), name, defaultValue);
+            return ResolveFieldContextExtensions.GetArgument(this, name, defaultValue);
         }
 
         public object GetArgument(System.Type argumentType, string name, object defaultValue = null)
         {
-            var argumentName = Schema?.FieldNameConverter.NameFor(name, null) ?? name;
-
-            if (Arguments == null || !Arguments.TryGetValue(argumentName, out var arg))
-            {
-                return defaultValue;
-            }
-
-            if (arg is Dictionary<string, object> inputObject)
-            {
-                if (argumentType == typeof(object))
-                    return arg;
-
-                if (argumentType.IsPrimitive())
-                    throw new InvalidOperationException($"Could not read primitive type '{argumentType.FullName}' from complex argument '{argumentName}'");
-
-                return inputObject.ToObject(argumentType);
-            }
-
-            var result = arg.GetPropertyValue(argumentType);
-
-            return result == null && argumentType.IsValueType ? defaultValue : result;
+            return ResolveFieldContextExtensions.GetArgument(this, argumentType, name, defaultValue);
         }
 
-        public bool HasArgument(string argumentName) => Arguments?.ContainsKey(argumentName) ?? false;
+        public bool HasArgument(string argumentName) => ResolveFieldContextExtensions.HasArgument(this, argumentName);
 
-        public Task<object> TryAsyncResolve(Func<ResolveFieldContext<TSource>, Task<object>> resolve, Func<ExecutionErrors, Task<object>> error = null)
+        public Task<object> TryAsyncResolve(Func<ResolveFieldContext, Task<object>> resolve, Func<ExecutionErrors, Task<object>> error = null)
```

Comment:
`TryAsyncResolve` and `TryAsyncResolve<TResult>` are used only in tests. Let's make them extension methods too. Then the `ResolveFieldContext` will become a pure state container.

Reply:
And I think that `Func<ResolveFieldContext, Task<object>> resolve` should be changed to `Func<IResolveFieldContext, Task<object>> resolve`

Reply:
Done